torchao/dtypes/channel_symmetricweight.py

```diff
@@ -0,0 +1,133 @@
+import torch
+from .uint4i import pack_uint4, unpack_uint4
+from .uint4i import UInt4Tensor
+from typing import Dict, Any
+
+SYMMETRIC_WEIGHT_OPS_TABLE: Dict[Any, Any] = {}
+
+def implements(aten_ops):
+    def decorator(fn):
+        for op in aten_ops:
+            SYMMETRIC_WEIGHT_OPS_TABLE[op] = fn
+        return fn
+    return decorator
+
+def _dynamically_quantize_per_channel_int4(x, quant_min, quant_max, target_dtype):
```

Comment:
also I feel this can probably be covered by AffineQuantizedTensor: https://github.com/pytorch/ao/blob/c2f9b84604536a72804787001c1b63daae792ee9/torchao/dtypes/affine_quantized_tensor.py#L86, and the previous `dequantize_per_channel` is calling our unified quant_primitive ops: https://github.com/pytorch/ao/blob/c2f9b84604536a72804787001c1b63daae792ee9/torchao/quantization/utils.py#L260

Reply:
Working on this